FinSearchUnified/Bundle/SearchBundleFindologic/ConditionHandler/SimpleConditionHandler.php

```diff
@@ -27,12 +29,12 @@ public function supportsCondition(ConditionInterface $condition)
      * Handles the passed condition object.
      *
      * @param ConditionInterface $condition
-     * @param QueryBuilder $query
+     * @param NewQueryBuilder $query
      * @param ShopContextInterface $context
      */
     public function generateCondition(
         ConditionInterface $condition,
-        QueryBuilder $query,
+        NewQueryBuilder $query,
```

Comment:
I'd suggest using the interface here.

Reply:
We do not have any interface for this. We have an abstract class, which is being used here. (Applicable to other comments below as well)